coderd/httpmw/ratelimit.go

```diff
@@ -22,14 +25,37 @@ func RateLimitPerMinute(count int) func(http.Handler) http.Handler {
 	}
 	return httprate.Limit(
 		count,
-		1*time.Minute,
+		window,
 		httprate.WithKeyFuncs(func(r *http.Request) (string, error) {
 			// Prioritize by user, but fallback to IP.
 			apiKey, ok := r.Context().Value(apiKeyContextKey{}).(database.APIKey)
-			if ok {
+			if !ok {
+				return httprate.KeyByIP(r)
+			}
+
+			if r.Header.Get(codersdk.BypassRatelimitHeader) == "" {
 				return apiKey.UserID.String(), nil
 			}
-			return httprate.KeyByIP(r)
+
+			// Allow Owner to bypass rate limiting for load tests
+			// and automation.
+			auth := UserAuthorization(r)
+
+			for _, role := range auth.Roles {
+				if role == rbac.RoleOwner() {
+					// HACK: use a random key each time to
+					// de facto disable rate limiting. The
+					// `httprate` package has no
+					// support for selectively changing the limit
+					// for particular keys.
+					return cryptorand.String(16)
+				}
+			}
```

Comment:
Another way to do this is add a resource for bypassing the rate limit. Then the FE can actually check if the capability exists, and we can give this to other roles as well.

```golang
ResourceRateLimitBypass = Object {
  Type: "bypass_rate_limit",
}
```

Reply:
Last I checked Rego was taking 20%+ of test CPU time, so it seems counterproductive the DoS prevention goals of the rate limiter. I added a comment to that effect.